File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid/Presenter/SharedView/SharedView_PopupControlBase.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Support.V7.App;
using Android.Views;
using Android.Widget;
using BPMOPMobile.Bean;
using BPMOPMobile.Droid.Class;
using Newtonsoft.Json.Linq;

namespace BPMOPMobile.Droid.Presenter.SharedView
{
    public class SharedView_PopupControlBase : SharedViewBase
    {
        public int flagView { get; set; }
        public ViewElement elementParent { get; set; }
        public ViewElement elementPopup { get; set; }  // Nếu là popup của control input grid detail mới xài cái này
        public JObject JObjectChild { get; set; } // Nếu là popup của control input grid detail mới xài cái này

        public SharedView_PopupControlBase(LayoutInflater _inflater, AppCompatActivity _mainAct, CustomBaseFragment _fragment, string _fragmentTag, View _rootView) : base(_inflater, _mainAct, _fragment, _fragmentTag, _rootView)
        {

        }

        /// <summary>
        /// Control động trên lưới Master
        /// </summary>
        public virtual void InitializeValue_Master(ViewElement elementParent)
        {
            flagView = (int)FlagViewControlDynamic.DetailWorkflow;

            this.elementParent = elementParent;
        }

        /// <summary>
        /// Control động trên popup lưới Detail
        /// </summary>
        public virtual void InitializeValue_InputGridDetail(ViewElement elementParent, ViewElement elementPopup, JObject JObjectChild)
        {
            flagView = (int)FlagViewControlDynamic.DetailWorkflow_InputGridDetail;

            this.elementParent = elementParent;
            this.elementPopup = elementPopup;
            this.JObjectChild = JObjectChild;
        }

        public virtual void InitializeValue_Master_CreateTask(ViewElement elementParent)
        {
            flagView = (int)FlagViewControlDynamic.DetailCreateTask;

            this.elementParent = elementParent;
        }

        public enum FlagViewControlDynamic
        {
            [Description("DetailWorkflow - Control")]
            DetailWorkflow = 0,

            [Description("DetailWorkflow - Control of popup InputGridDetail")]
            DetailWorkflow_InputGridDetail = 1,

            [Description("DetailCreateTask - Control")]
            DetailCreateTask = 2,
        }
    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/shareview/SharedView_PopupControlBase.java
package com.vuthao.bpmop.shareview;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;

import com.vuthao.bpmop.base.model.dynamic.ViewElement;
import com.vuthao.bpmop.core.Vars;

import org.json.JSONObject;

public class SharedView_PopupControlBase extends SharedViewBase{
    public int flagView;
    public ViewElement elementParent;
    public ViewElement elementPopup;  // Nếu là popup của control input grid detail mới xài cái này
    public JSONObject JObjectChild; // Nếu là popup của control input grid detail mới xài cái này

    public SharedView_PopupControlBase(LayoutInflater inflater, Activity mainAct, String fragmentTag, View rootView) {
        super(inflater, mainAct, fragmentTag, rootView);
    }

    public void initializeValue_Master(ViewElement viewElement) {
        flagView = Vars.FlagViewControlDynamic.DetailWorkflow;
        this.elementParent = viewElement;
    }

    public void initializeValue_InputGridDetail(ViewElement elementParent, ViewElement elementPopup, JSONObject JObjectChild) {
        flagView = Vars.FlagViewControlDynamic.DetailWorkflow_InputGridDetail;
        this.elementParent = elementParent;
        this.elementPopup = elementPopup;
        this.JObjectChild = JObjectChild;
    }

    public void initializeValue_Master_CreateTask(ViewElement elementParent) {
        flagView = Vars.FlagViewControlDynamic.DetailCreateTask;
        this.elementParent = elementParent;
    }
}
